starlette_feedgen/utils.py

```diff
@@ -1,35 +1,145 @@
+                return prefix + ":" + name[1]
+        # Return the unqualified name
+        return name[1]
+
+    async def _finish_pending_start_element(self):
+        if self._pending_start_element:
+            await self._write('>')
+            self._pending_start_element = False
+
+    # ContentHandler methods
+
+    async def startDocument(self):
+        await self._write('<?xml version="1.0" encoding="%s"?>\n' %
+                    self._encoding)
+
+    async def endDocument(self):
+        await self._flush()
+
+    def startPrefixMapping(self, prefix, uri):
+        self._ns_contexts.append(self._current_context.copy())
+        self._current_context[uri] = prefix
+        self._undeclared_ns_maps.append((prefix, uri))
+
+    def endPrefixMapping(self):
+        self._current_context = self._ns_contexts[-1]
+        del self._ns_contexts[-1]
+
+    async def startElement(self, name, attrs):
+        await self._finish_pending_start_element()
+        await self._write('<' + name)
+        for (name, value) in attrs.items():
+            await self._write(' %s=%s' % (name, quoteattr(value)))
+        if self._short_empty_elements:
+            self._pending_start_element = True
+        else:
+            await self._write(">")
+
+    async def endElement(self, name):
+        if self._pending_start_element:
+            await self._write('/>')
+            self._pending_start_element = False
+        else:
+            await self._write('</%s>' % name)
+
+    async def startElementNS(self, name, attrs):
+        await self._finish_pending_start_element()
+        await self._write('<' + self._qname(name))
+
+        for prefix, uri in self._undeclared_ns_maps:
+            if prefix:
+                await self._write(' xmlns:%s="%s"' % (prefix, uri))
+            else:
+                await self._write(' xmlns="%s"' % uri)
+        self._undeclared_ns_maps = []
+
+        for (name, value) in attrs.items():
+            await self._write(' %s=%s' % (self._qname(name), quoteattr(value)))
+        if self._short_empty_elements:
+            self._pending_start_element = True
+        else:
+            await self._write(">")
+
+    async def endElementNS(self, name):
+        if self._pending_start_element:
+            await self._write('/>')
+            self._pending_start_element = False
+        else:
+            await self._write('</%s>' % self._qname(name))
+
+    async def characters(self, content):
+        if content:
+            await self._finish_pending_start_element()
+            if not isinstance(content, str):
+                content = str(content, self._encoding)
+            await self._write(escape(content))
+
+    async def ignorableWhitespace(self, content):
+        if content:
+            await self._finish_pending_start_element()
+            if not isinstance(content, str):
+                content = str(content, self._encoding)
```

Comment:
Do you mean bytes object? In this case i suggest to write like this:

```
if isinstance(content, bytes):
    content = content.decode(self._encoding)
await self._write(content)
```

Reply:
the thing here is that we just grabbed the source code from standard library `xml.sax.saxutils` and rewrote it in async way :) it seems like in our implementation `content` is always string,  but i would remain that `if` just in case and use your snippet to be more accurate, thx